fs-storage/src/base_storage.rs

```diff
@@ -0,0 +1,41 @@
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::str::FromStr;
+
+use data_error::Result;
+
+pub trait BaseStorage<K, V>
+where
+    K: FromStr + Hash + Eq + Ord + Debug + Clone,
+    V: Debug + Clone,
+{
+    fn get(&self, id: &K) -> Option<V>;
+    fn set(&mut self, id: K, value: V) -> Result<()>;
+    fn remove(&mut self, id: &K) -> Result<()>;
+
+    /// Remove file at stored path
+    fn erase(&mut self) -> Result<()>;
```

Comment:
I don't think we need mutable reference here 
```suggestion
    fn erase(&self) -> Result<()>;
```